### ISSUE TITLE
Question about syncing entire remote pool, *without* mounting dangerous bits

### ISSUE THREAD

**Issue by hartzell** [jimsalterjrs/sanoid#403]
Hi All,

**TL;DR** -- I'm concerned about remote filesystems that I pull onto my backup server being mounted based on their mountpoint option (e.g. `/usr`), which would screw up the backup server.  I can't figure out whether `syncoid` is "safe-for-george" out of the box or whether I need to use "power user options" (e.g. from #273) or .....

---

I have a remote FreeBSD 12 system, using the bog-standard installer options for a ZFS system, plus a few extra ZFS filesystems for poudriere and etc....

```consolesession
% zfs list
NAME                                                 USED  AVAIL  REFER  MOUNTPOINT
zroot                                               22.0G  47.3G    88K  /zroot
zroot/ROOT                                          7.00G  47.3G    88K  none
zroot/ROOT/default                                  7.00G  47.3G  6.57G  /
zroot/poudriere                                     1.80G  47.3G    88K  /zroot/poudriere
zroot/poudriere/data                                 543M  47.3G    96K  /usr/local/poudriere/data
zroot/poudriere/data/.m                              200K  47.3G    88K  /usr/local/poudriere/data/.m
zroot/poudriere/data/cache                          4.73M  47.3G  4.57M  /usr/local/poudriere/data/cache
zroot/poudriere/data/images                           88K  47.3G    88K  /usr/local/poudriere/data/images
zroot/poudriere/data/logs                           9.64M  47.3G  9.16M  /usr/local/poudriere/data/logs
zroot/poudriere/data/packages                        528M  47.3G   528M  /usr/local/poudriere/data/packages
zroot/poudriere/data/wrkdirs                          88K  47.3G    88K  /usr/local/poudriere/data/wrkdirs
zroot/poudriere/jails                               1.26G  47.3G    88K  /zroot/poudriere/jails
zroot/poudriere/jails/12_0                          1.26G  47.3G  1.26G  /usr/local/poudriere/jails/12_0
zroot/poudriere/ports                                596K  47.3G    88K  /zroot/poudriere/ports
zroot/poudriere/ports/ports_and_hartzell             452K  47.3G   695M  /usr/local/poudriere/ports/ports_and_hartzell
zroot/tmp                                           2.54M  47.3G   328K  /tmp
zroot/usr                                           13.2G  47.3G    88K  /usr
zroot/usr/home                                      12.5G  47.3G  12.1G  /usr/home
zroot/usr/ports                                       88K  47.3G    88K  /usr/ports
zroot/usr/portshaker                                 711M  47.3G    88K  /usr/portshaker
zroot/usr/portshaker/github_hartzell_freebsd-ports   364K  47.3G   276K  /usr/portshaker/github_hartzell_freebsd-ports
zroot/usr/portshaker/ports                           710M  47.3G   695M  /usr/portshaker/ports
zroot/usr/src                                         88K  47.3G    88K  /usr/src
zroot/var                                           11.3M  47.3G    88K  /var
zroot/var/audit                                       88K  47.3G    88K  /var/audit
zroot/var/crash                                       88K  47.3G    88K  /var/crash
zroot/var/log                                       5.98M  47.3G  1.21M  /var/log
zroot/var/mail                                      4.54M  47.3G   560K  /var/mail
zroot/var/tmp                                        480K  47.3G    88K  /var/tmp
```

I'd love to just mirror the entire remote `zroot` pool beneath a filesystem on my backup server (e.g. `storage/backups/the-remote-system`).

But....  A handful of filesystems (e.g. `/usr`, *etc*...) are mounted in specific locations via the mountpoint option:

```consolesession
% zfs get -t filesystem mountpoint
NAME                                                PROPERTY    VALUE                                          SOURCE
zroot                                               mountpoint  /zroot                                         local
zroot/ROOT                                          mountpoint  none                                           local
zroot/ROOT/default                                  mountpoint  /                                              local
zroot/poudriere                                     mountpoint  /zroot/poudriere                               inherited from zroot
zroot/poudriere/data                                mountpoint  /usr/local/poudriere/data                      local
zroot/poudriere/data/.m                             mountpoint  /usr/local/poudriere/data/.m                   inherited from zroot/poudriere/data
zroot/poudriere/data/cache                          mountpoint  /usr/local/poudriere/data/cache                inherited from zroot/poudriere/data
zroot/poudriere/data/images                         mountpoint  /usr/local/poudriere/data/images               inherited from zroot/poudriere/data
zroot/poudriere/data/logs                           mountpoint  /usr/local/poudriere/data/logs                 inherited from zroot/poudriere/data
zroot/poudriere/data/packages                       mountpoint  /usr/local/poudriere/data/packages             inherited from zroot/poudriere/data
zroot/poudriere/data/wrkdirs                        mountpoint  /usr/local/poudriere/data/wrkdirs              inherited from zroot/poudriere/data
zroot/poudriere/jails                               mountpoint  /zroot/poudriere/jails                         inherited from zroot
zroot/poudriere/jails/12_0                          mountpoint  /usr/local/poudriere/jails/12_0                local
zroot/poudriere/ports                               mountpoint  /zroot/poudriere/ports                         inherited from zroot
zroot/poudriere/ports/ports_and_hartzell            mountpoint  /usr/local/poudriere/ports/ports_and_hartzell  local
zroot/tmp                                           mountpoint  /tmp                                           local
zroot/usr                                           mountpoint  /usr                                           local
zroot/usr/home                                      mountpoint  /usr/home                                      inherited from zroot/usr
zroot/usr/ports                                     mountpoint  /usr/ports                                     inherited from zroot/usr
zroot/usr/portshaker                                mountpoint  /usr/portshaker                                inherited from zroot/usr
zroot/usr/portshaker/github_hartzell_freebsd-ports  mountpoint  /usr/portshaker/github_hartzell_freebsd-ports  inherited from zroot/usr
zroot/usr/portshaker/ports                          mountpoint  /usr/portshaker/ports                          inherited from zroot/usr
zroot/usr/src                                       mountpoint  /usr/src                                       inherited from zroot/usr
zroot/var                                           mountpoint  /var                                           local
zroot/var/audit                                     mountpoint  /var/audit                                     inherited from zroot/var
zroot/var/crash                                     mountpoint  /var/crash                                     inherited from zroot/var
zroot/var/log                                       mountpoint  /var/log                                       inherited from zroot/var
zroot/var/mail                                      mountpoint  /var/mail                                      inherited from zroot/var
zroot/var/tmp                                       mountpoint  /var/tmp                                       inherited from zroot/var
hartzell@corvid:/usr/local/mailman %
```

Will syncoid take care of this "out of the box" for me?  Do I need to use some poweruser option, e.g. from #273?    

*FreeBSD* (the remote system) does not seem to support the `-o` or `-x` options for `zfs receive`, but *FreeNAS* is ["downstream of Illumos" for its ZFS](https://www.reddit.com/r/freenas/comments/be1ah5/freebsd_112_eol_and_effects_for_freenas/el3o7u7/) and does seem to support them.

Is there a standard example of a "mirror a remote system" configuration or docs that I should peruse?

Thanks for pointers!

**Comment by phreaker0** [comment]
Syncoid doesn't mangle properties by default. You can pass zfs recv the -u option so it will not mount any received replication stream. But you need to take care when you reimport the pool on the backup server. If it's a sole backup pool you could always import it without mounting anything.
The syncoid command would look like:
`syncoid -r --recvoptions=u zroot root@backup:storage/backups/the-remote-system`

I myself set all the mountpoints on the backup server to none after the initial replication (with -u) so it never tries to mount them but you loose all the mountpoint information.

**Comment by hartzell** [comment]
Thanks, that seems to work.

Is there some part of `syncoid` that's supposed to be deleting snapshots on the backup server after they've been deleted (by some other process) on the remote system?

**Comment by phreaker0** [comment]
syncoid doesn't cleanup snapshots (except for the one itself creates). So the backup server needs to do this on it's own. If you run sanoid on the source, you can run sanoid on the backup tool for pruning only.
